src/nashEquilibria.jl

```diff
@@ -0,0 +1,207 @@
+#=
+Tools for computing Nash equilibria of normal form games.
+
+Includes: Pre-processing, Lemke-Howson, Simple Search.
+
+Authors: Arnav Sood
+=#
+
+using NonNegLeastSquares
+include("Games.jl")
+include("normal_form_game.jl")
+
+"""
+Render a n-player normal form game non-degenerate by adding a suitable positive constant to each payoff (i.e., by ensuring each payoff is strictly greater than 0).
+
+##### Arguments
+
+- `g::NormalFormGame` : A NormalFormGame object.
+
+##### Returns
+
+- `g::NormalFormGame` : A NormalFormGame object.
+"""
+function unDegenerate!(g::NormalFormGame)
+    # Get the normalizing constant.
+    numPlayers = num_players(g)
+    minArray = Array(Real, numPlayers)
+    for i in 1:numPlayers
+         minArray[i] = minimum(g.players[i].payoff_array)
+    end
+    normalizer = abs(minimum(minArray)) + 1
+
+    # Alter the payoff matrices.
+    for i in 1:numPlayers
+        g.players[i].payoff_array += normalizer
+    end
+    return g
+end
+
+# This is currently very slow; i.e., each call removes one dominated strategy for one player.
+
+"""
+Simplify a n-player normal form game by the iterated removal of dominated strategies.
+
+##### Arguments
+
+- `g::NormalFormGame` : A NormalFormGame object.
+
+##### Returns
+
+- `g::NormalFormGame` : A NormalFormGame object.
+"""
+function unDominate!(g::NormalFormGame)
+    changeFlag = true
+    changesMade = false
+    while changeFlag == true
+        changeFlag = unDominate_Auxiliary!(g::NormalFormGame)
```

Comment:
I don't think you need to do ```g::NormalFormGame``` since you already specified in the function definition that g is a `NormalFormGame`